packages/hub/services/search-index.ts

```diff
@@ -240,6 +243,17 @@ class IndexerRun implements IndexerHandle {
     } while (rows.length > 0);
   }
 
+  async read(card: RawCard): Promise<RawCard | void> {
```

Comment:
Instead of `read`, let's call this `reindex` and give it a signature like:

```js
async reindex(card: CardId, action: 'save' | 'delete'): Promise<void>;
```

Then the whole switch statement from `indexCardFromNotification` would move inside `reindex`.